File: Algorithm/Level2/전화번호 목록/solution.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    int size_book = phone_book.size();
    
    sort(phone_book.begin(), phone_book.end());
    
    for(int i = 0; i < size_book; i++){
        if(i+1 >= size_book)
            return answer;
        if(phone_book[i].find(phone_book[i+1].substr(0,phone_book[i].length())) == 0){
            answer = false;
            return answer;
       }
    }
}

File: Algorithm/Level2/전화번호 목록/solution.py
def solution(phone_book)
    answer = True
    phone_book.sort()
    for i in range(1, len(phone_book))
        len_text = len(phone_book[i-1])
        print(phone_book[i][0:len_text])
        if phone_book[i-1] == phone_book[i][0:len_text]
            return False
    return answer